### PR TITLE
[TW-396] Update ForageConfig and setForageConfig kDoc comments

### PR DESCRIPTION
**Pull request by kimberleehowley**
## What

This PR updates comments for ForageConfig and setForageConfig. 

## Why

Improving documentation hopefully makes the SDK easier to use. 

## Test Plan

- ❌ |  No unit tests, because this PR only changes comments. 
- ✅ |  The reviewer should manually test the changes in this PR. 

To test, checkout my branch and run: 

```
.gradlew dokkaHtml
npx http-serve reference-docs
```

## Demo

Screenshot 2024-01-24 at 10 41 03 AM

Screenshot 2024-01-24 at 10 40 49 AM

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt [hunk 1]
```diff
@@ -24,6 +35,13 @@ data class ForageConfig(
 interface ForageElement<T : ElementState> {
     var typeface: Typeface?
 
+    /**
+     * Sets the necessary [ForageConfig] configuration properties for a [ForageElement].
+     *
+     * [setForageConfig] must be called before any other methods can be executed on the Element.
+     *
+     * @param params A [ForageConfig] instance that specifies a `merchantId` and `sessionToken`.
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt:43]: In this case the parameter is called `forageConfig`, as seen in the line below
```suggestion
     * @param forageConfig A [ForageConfig] instance that specifies a `merchantId` and `sessionToken`.
```

File: forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt [hunk 2]
```diff
@@ -6,10 +6,21 @@ import com.joinforage.forage.android.core.element.StatefulElementListener
...
  *
- * @property merchantId Your merchant FNS number
- * @property sessionToken A temporary token used for authenticate
+ * Pass a [ForageConfig] instance in a call to
+ * [setForageConfig][com.joinforage.forage.android.ui.AbstractForageElement.setForageConfig] to
+ * configure an Element.
+ *
+ * @property merchantId Either a unique seven digit numeric string that
+ * [FNS](https://docs.joinforage.app/docs/ebt-online-101#food-and-nutrition-service-fns) issues
+ * to authorized EBT merchants, or a unique merchant ID that Forage provides during onboarding.
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt:17]: Would be good to provide an example, maybe using the Kdoc `@sample` or some other means?

File: forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt [hunk 3]
```diff
@@ -6,10 +6,21 @@ import com.joinforage.forage.android.core.element.StatefulElementListener
...
- * Configuration details required by ForageElement to operate correctly
+ * The configuration details that Forage needs to create a functional [ForageElement].
  *
- * @property merchantId Your merchant FNS number
- * @property sessionToken A temporary token used for authenticate
+ * Pass a [ForageConfig] instance in a call to
+ * [setForageConfig][com.joinforage.forage.android.ui.AbstractForageElement.setForageConfig] to
+ * configure an Element.
+ *
+ * @property merchantId Either a unique seven digit numeric string that
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt:15]: Can we emphasize the `mid/<merchant-id>` format (from the Forage dashboard) and mention that the ID can be retrieved from the dashboard?

The accepted formats are:

- `mid/<merchant-id>` (has to be prefixed with `mid/`)
- `<fns-number>`

**Review by djoksimo — CHANGES_REQUESTED** [review]
Nice stuff @kimberleehowley ! Left some comments regarding merchant Id and a typo fix

Would be good for @devinmorgan to verify this one given that he introduced `ForageConfig` and `setForageConfig`

File: forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt (hunk 2 above)
Comment (kimberleehowley) [reply forage-android/src/main/java/com/joinforage/forage/android/ui/ForageElement.kt:17]: @djoksimo In the last PR, I struggled with formatting issues with `@sample`. Is it okay with you if I just do them all at once at the end? 

I created a [backlog ticket](https://linear.app/joinforage/issue/TW-415/add-sample-code-tags-after-the-quickstart-has-been-approved-and-other) for this task during the last PR.

**Review by djoksimo — APPROVED** [review]
LGTM!

**Comment by devinmorgan** [comment]
This PR looks good! Nice work all!

**Pull request merged**